src/Fixer/FunctionNotation/NullableTypeHintForDefaultNullValueFixer.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\FunctionNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author HypeMC
+ */
+final class NullableTypeHintForDefaultNullValueFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Adds or removes `?` before type hints for parameters with a default `null` value.',
+            [
+                new VersionSpecificCodeSample(
+                    "<?php\nfunction sample(string \$str = null)\n{}\n",
+                    new VersionSpecification(70100)
+                ),
+                new VersionSpecificCodeSample(
+                    "<?php\nfunction sample(?string \$str = null)\n{}\n",
+                    new VersionSpecification(70100),
+                    ['use_nullable_type_hint' => false]
+                ),
+            ],
+            'Rule is applied only in a PHP 7.1+ environment.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return \PHP_VERSION_ID >= 70100 && $tokens->isAllTokenKindsFound([T_FUNCTION, T_VARIABLE]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // Should run after CompactNullableTypehintFixer and before MethodArgumentSpaceFixer
+        return -1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('use_nullable_type_hint', 'Whether to add or remove `?` before type hints for parameters with a default `null` value.'))
```

Comment:
I would rename the option to `use_nullable_type_declaration`.

In the description, _type hints_ should be _type declarations_.

Reply:
Done